zebra-rpc/src/methods.rs

```diff
@@ -861,65 +839,55 @@ where
-                        zebra_state::ReadResponse::SaplingTree(None) => 0,
-                        _ => unreachable!("unmatched response to a SaplingTree request"),
-                    };
-
                 let orchard_tree_response = futs.next().await.expect("`futs` should not be empty");
-                let orchard_note_commitment_tree_count =
-                    match orchard_tree_response.map_server_error()? {
-                        zebra_state::ReadResponse::OrchardTree(Some(nct)) => nct.count(),
-                        zebra_state::ReadResponse::OrchardTree(None) => 0,
-                        _ => unreachable!("unmatched response to a OrchardTree request"),
-                    };
-
-                // From <https://zcash.github.io/rpc/getblock.html>
-                const NOT_IN_BEST_CHAIN_CONFIRMATIONS: i64 = -1;
-
-                let depth_response = futs.next().await.expect("`futs` should not be empty");
-                let confirmations = match depth_response.map_server_error()? {
-                    // Confirmations are one more than the depth.
-                    // Depth is limited by height, so it will never overflow an i64.
-                    zebra_state::ReadResponse::Depth(Some(depth)) => i64::from(depth) + 1,
-                    zebra_state::ReadResponse::Depth(None) => NOT_IN_BEST_CHAIN_CONFIRMATIONS,
-                    _ => unreachable!("unmatched response to a depth request"),
+                let zebra_state::ReadResponse::OrchardTree(orchard_tree) =
+                    orchard_tree_response.map_server_error()?
+                else {
+                    unreachable!("unmatched response to a OrchardTree request");
                 };
 
-                let (time, height) = if should_read_block_header {
-                    let block_header_response =
-                        futs.next().await.expect("`futs` should not be empty");
+                let nu5_activation = NetworkUpgrade::Nu5.activation_height(&self_clone.network);
 
-                    match block_header_response.map_server_error()? {
-                        zebra_state::ReadResponse::BlockHeader { header, height, .. } => {
-                            (Some(header.time.timestamp()), Some(height))
-                        }
-                        _ => unreachable!("unmatched response to a BlockHeader request"),
-                    }
-                } else {
-                    (None, hash_or_height.height())
-                };
+                // This could be `None` if there's a chain reorg between state queries.
+                let orchard_tree =
+                    orchard_tree.ok_or_server_error("missing orchard tree for block")?;
+
+                let final_orchard_root =
+                    if nu5_activation.is_some() && height >= nu5_activation.unwrap() {
+                        Some(orchard_tree.root().into())
+                    } else {
+                        None
+                    };
```

Comment:
Optional/Nitpick: We could avoid the `unwrap()` here with a match statement.

```suggestion
                let final_orchard_root = match nu5_activation {
                    Some(activation_height) if height >= activation_height => {
                        Some(orchard_tree.root().into())
                    }
                    _other => None,
                };
```